gremlin-go/example/example.go

```diff
@@ -22,37 +22,128 @@ package main
 import (
 	"fmt"
 	gremlingo "github.com/apache/tinkerpop/gremlin-go/driver"
-	"github.com/apache/tinkerpop/gremlin-go/v3/driver"
 )
 
 // syntactic sugar
 var __ = gremlingo.T__
 var gt = gremlingo.P.Gt
 var order = gremlingo.Order
+var T = gremlingo.T
+var P = gremlingo.P
 
 func main() {
-	// Creating the connection to the server.
-	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin",
+	connectionExample()
+	basicGremlinExample()
+	modernTraversalExample()
+}
+
+func connectionExample() {
+	// Creating the connection to the server
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin")
+
+	// Connecting to the server with customized configurations
+	driverRemoteConnection, err = gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin",
 		func(settings *gremlingo.DriverRemoteConnectionSettings) {
-			settings.TraversalSource = "gmodern"
+			settings.TraversalSource = "g"
+			settings.NewConnectionThreshold = 4
+			settings.EnableCompression = false
+			settings.ReadBufferSize = 0
+			settings.WriteBufferSize = 0
 		})
+
+	// Error handling
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
 	// Cleanup
 	defer driverRemoteConnection.Close()
 
-	// Creating graph traversal
+	// Creating the graph traversal
 	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
+	fmt.Println(g)
+}
 
-	// Perform traversal
-	result, err := g.V().HasLabel("person").Has("age", __.Is(gt(28))).Order().By("age", order.Desc).Values("name").ToList()
+func basicGremlinExample() {
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin")
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
-	for _, r := range result {
-		fmt.Println(r.GetString())
+	defer driverRemoteConnection.Close()
+	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
+
+	// Basic Gremlin: adding and retrieving data
+	v1, err := g.AddV("person").Property("name", "marko").Next()
+	v2, err := g.AddV("person").Property("name", "stephen").Next()
+	v3, err := g.AddV("person").Property("name", "vadas").Next()
+	v1Vertex, err := v1.GetVertex()
+	v2Vertex, err := v2.GetVertex()
+	v3Vertex, err := v3.GetVertex()
+
+	// Be sure to use a terminating step like next() or iterate() so that the traversal "executes"
+	// Iterate() does not return any data and is used to just generate side-effects (i.e. write data to the database)
+	g.V(v1Vertex).AddE("knows").To(v2Vertex).Property("weight", 0.75).Iterate()
```

Comment:
`.Iterate()` return channel, so need to wait for result

```suggestion
	promise := g.V(v1Vertex).AddE("knows").To(v2Vertex).Property("weight", 0.75).Iterate()
	err = <-promise 
	// ... error handling 
```

Reply:
Added waits and error handling as suggested